traveler-ios-ui/TravelerKitUI/ViewControllers/ParkingResultMapViewController.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  ParkingResultMapViewController.swift
+//  TravelerKitUI
+//
+//  Created by Ata Namvari on 2019-10-08.
+//  Copyright © 2019 Guestlogix Inc. All rights reserved.
+//
+
+import Foundation
+import TravelerKit
+import MapKit
+
+let parkingAnnotationIdentifier = "parkingAnnotationIdentifier"
+
+protocol ParkingResultMapViewControllerDelegate: class {
+    func parkingResultMapViewController(_ controller: ParkingResultMapViewController, didChange boundingBox: BoundingBox)
+}
+
+class ParkingResultMapViewController: UIViewController {
+    @IBOutlet weak var mapView: MKMapView!
+
+    var context: ParkingResultContext?
+    weak var delegate: ParkingResultMapViewControllerDelegate?
+
+    private var isSelecting = false
+    private var queryUpdated = false
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        mapView.register(LabeledAnnotationView.self, forAnnotationViewWithReuseIdentifier: parkingAnnotationIdentifier)
+
+        context?.addObserver(self)
+    }
+
+    deinit {
+        context?.removeObserver(self)
+    }
+}
+
+extension ParkingResultMapViewController: ParkingResultContextObserving {
+    func parkingResultContextDidChangeSelectedIndex(_ context: ParkingResultContext) {
+        guard let index = context.selectedIndex else {
+            return
+        }
+
+        isSelecting = true
```

Comment:
Is the `isSelecting` property actually necessary?

Reply:
Yes its being used in the mapView didSelect